### ISSUE TITLE
macOS runtime error

### ISSUE THREAD

**Issue by g5t** [pace-neutrons/libpymcr#6]
## Issue
When loading a `.ctf` file on a macOS machine,
```
Main must be started using mclRunMain
```
is issued and the Python interpreter exits with error code `1`.

## Background
The purpose of the missing call to `mclRunMain` is to ["handle specific threading issues on the Mac OS platform."](https://www.mathworks.com/matlabcentral/answers/97697-what-is-the-purpose-of-mclrunmain-in-the-matlab-compiler-matrixdriver-example)
The same functionality is also [accessible via the `cpplib` interface](https://www.mathworks.com/help/compiler_sdk/cxx/matlab.cpplib.runmain.html) used by `libpymcr`. 

## Problem
The offending call within `libpymcr` occurs within the `matlab_env` constructor at the call to `matlab::cpplib::initMATLABLibrary`.

## Possible solution
Following the [example for use of `initMATLABLibrary`](https://www.mathworks.com/help/compiler_sdk/cxx/matlab.cpplib.initmatlablibrary.html), adding the option `-nojvm` to the call to `matlab::cpplib::initMATLABApplication` allows the library to initialize without the error.

### Shortcoming of this solution
A simple test library, [`hello.ctf`.tar.gz](https://github.com/pace-neutrons/libpymcr/files/10850271/hello.ctf.tar.gz)
exposes a single function `hello()` in addition to the required `call` function. 
The `hello` function takes no arguments, prints "Hello World", and returns no arguments.

Without the addition of the `-nojvm` argument, MATLAB fails to initialize with the error noted above.
With the addition of the option, the initialization proceeds and the function can be called, e.g.,

```python
import libpymcr
m = libpymcr.Matlab('hello.ctf')
m.hello()
```
On first-call the function prints "Hello World" as anticipated, but not for subsequent calls on a macOS system. (The same function works with or without the `-nojvm` option on a Linux machine on the first and subsequent calls.)

## Alternative
It may be possible to use the `RunMain` interface without needing to specify `-nojvm`, but this has not been investigated.

**Comment by g5t** [comment]
I've updated my possible solution, #7, to allow a user to set the `initMATLABApplication` `options` vector from the Python `libpymcr.Matlab` initializer.

Now, by default, `'-nojvm'` is included in the options for macOS systems. On all other platforms the options remain empty.
This should have the result that library `ctf` files can be loaded on all systems but MATLAB GUI interfaces are not available under macOS.

**Comment by mducle** [comment]
Thanks Greg. I've tried to use `runMain` but keep getting segfaults when it exits. I think internally it destroys either the app or the library when it exits even if the actual C++ code we write doesn't (the `MATLABApplication` class is just a wrapper that calls a C API function `terminate_mvm_instance` on exit and doesn't do anything else; `initMATLABApplication` calls the C API `runtime_create_session`, and `runMain` calls `cppshared_run_main`). I tried to call `cppshared_run_main` directly but I get a `Info.plist not found` error on Mac, and it still segfaults on Linux. :(

**Comment by g5t** [comment]
Given the difficulties with `runMain`, etc., is my new proposed solution acceptable (at least for now)? 

With shared-memory between MATLAB and Python, maybe it wouldn't be _too_ hard to write a `Matplotlib` approximation for the PACE/Horace plotting routines at some point in the future? And until then macOS users would need to roll-their-own plotting or live without.

**Comment by mducle** [comment]
Yeah, that's my thought too. I was thinking of two components - a Matlab wrapper which overrides the Matlab `plot` command and plot handle objects which would capture user input arguments and then pass it back to Python, and a Python class which then translates this to matplotlib - this would also enable better integration for Jupyter as well since the Matlab graphics doesn't work at all with the various javascript widgets often used there.

So I'm happy with the PR - do you want to add anything else or can I merge it?

**Comment by g5t** [comment]
This was fixed by #7 -- but there are apparently more issues with running under macOS
